### PR TITLE
doc: api: Reference MQTT over sockets library

### PR DESCRIPTION
**Pull request by rlubos**
The MQTT over sockets doxygen group was not referenced, hence the API
documentation was missing. Replace legacy MQTT lib reference with a new
one.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

### CODE REVIEW

**Review by dbkinder — APPROVED** [review]
LGTM, thanks.

File: doc/api/networking.rst [hunk 1]
```diff
@@ -192,10 +192,10 @@ DHCPv4
 .. doxygengroup:: dhcpv4
    :project: Zephyr
 
-MQTT 3.1.1
-==========
+MQTT 3.1.1 over Sockets
```
Comment (nashif) [inline doc/api/networking.rst]: Do we want to keep referring to this as MQTT over sockets, also when the old implementation is going to be removed? Can we just call this MQTT and drop the socket from the name of the doxygen group and folders?

**Review by nashif — CHANGES_REQUESTED** [review]

File: doc/api/networking.rst (hunk 1 above)
Comment (rlubos) [reply doc/api/networking.rst]: I think we can drop the `over Sockets` part, just added it to be consistent with CoAP. @jukkar what do you think?

Comment (jukkar) [reply doc/api/networking.rst]: Yes, we should drop the socket from various names etc. Can I propose that we remove the legacy MQTT first, and then do these renames etc in separate PR?

**Comment by rlubos** [comment]
Removed the `over Sockets` part.

**Review by nashif — APPROVED** [review]

**Pull request merged**